tests/beta/tapes/test_qnode.py

```diff
@@ -0,0 +1,429 @@
+            QNode._validate_backprop_method(dev, None)
+
+        test_interface = "something"
+        monkeypatch.setitem(dev._capabilities, "passthru_interface", test_interface)
+
+        with pytest.raises(QuantumFunctionError, match=f"when using the {test_interface}"):
+            QNode._validate_backprop_method(dev, None)
+
+        tape_class, interface, method = QNode._validate_backprop_method(dev, test_interface)
+
+        assert tape_class is QuantumTape
+        assert method == "backprop"
+        assert interface == None
+
+    def test_get_parameter_shift_method(self, monkeypatch):
+        """Test that the get_parameter_shift_method method correctly
+        returns the correct tape."""
+
+        # TODO: uncomment the following block when the parameter shift
+        # tape is implemented.
+
+        # qubit devices return the QubitParamShiftTape
+        # dev = qml.device("default.qubit", wires=1)
+        # tape_class, interface, method = QNode._get_parameter_shift_method(dev, "autograd")
+        # assert tape_class is QubitParamShiftTape
+        # assert method == "analytic"
+        # assert interface == "autograd"
+
+
+        # Qubit devices not yet supported; raises a warning and returns the
+        # finite difference method
+        dev = qml.device("default.qubit", wires=1)
+
+        with pytest.warns(UserWarning, match="Qubit parameter-shift rule not yet implemented"):
+            tape_class, interface, method = QNode._get_parameter_shift_method(dev, "autograd")
+
+        assert tape_class is QuantumTape
+        assert method == "numeric"
+        assert interface == "autograd"
+
+        # CV devices not yet supported; raises a warning and returns the
+        # finite difference method
+        dev = qml.device("default.gaussian", wires=1)
+
+        with pytest.warns(UserWarning, match="CV parameter-shift rule not yet implemented"):
+            tape_class, interface, method = QNode._get_parameter_shift_method(dev, "autograd")
+
+        assert tape_class is QuantumTape
+        assert method == "numeric"
+        assert interface == "autograd"
+
+        # test an unknown model raises an exception
+
+        def capabilities(cls):
+            capabilities = cls._capabilities
+            capabilities.update(model="None")
+            return capabilities
+
+        monkeypatch.setattr(qml.devices.DefaultQubit, "capabilities", capabilities)
+        dev = qml.device("default.qubit", wires=1)
+
+        with pytest.raises(QuantumFunctionError, match="does not support the parameter-shift rule"):
+            QNode._get_parameter_shift_method(dev, None)
+
+    def test_best_method(self, monkeypatch):
+        """Test that the method for determining the best diff method
+        for a given device and interface works correctly"""
+        dev = qml.device("default.qubit", wires=1)
+        monkeypatch.setitem(dev._capabilities, "passthru_interface", 1)

```

Comment:
The value `1` is slightly confusing to me here :thinking:

Reply:
good idea!